tests/core/test_imagedata_filters.py

```diff
@@ -166,6 +167,180 @@ def test_contour_labeled_with_invalid_scalars():
+    #   is otherwise surrounded by background,
+
+    # Second label:
+    #   has two points near center of image,
+    #   is adjacent to first label,
+    #   has one side touching image boundary,
+    #   is otherwise surrounded by background
+
+    dim = (4, 3, 3)
+    labels = np.zeros(np.prod(dim))
+    labels[17] = 2  # First label
+    labels[[18, 19]] = 5  # Second label
+    image = pv.ImageData(dimensions=dim)
+    image.point_data['labels'] = labels
+
+    label_ids = np.unique(image.point_data.active_scalars).tolist()
+    assert label_ids == [0, 2, 5]
+    return image
+
+
+def _get_label_ids(array):
+    """Get unique foreground label ids."""
+    ids = np.unique(array).tolist()
+    ids.remove(0) if 0 in ids else None
+    return ids
+
+
+def _get_ids_with_background_boundary(mesh):
+    """Return region ids from the boundary_labels array which share a boundary with the background."""
+    extracted = mesh.extract_values(0, component_mode=1)
+    external_polygons = extracted[BOUNDARY_LABELS] if BOUNDARY_LABELS in extracted.cell_data else []
+    return _get_label_ids(external_polygons)
+
+
+def _get_ids_with_internal_boundary(mesh):
+    """Return region ids from the boundary_labels array which share internal boundaries."""
+    extracted = mesh.extract_values(0, invert=True, component_mode=1)
+    internal_polygons = extracted[BOUNDARY_LABELS] if BOUNDARY_LABELS in extracted.cell_data else []
+    return _get_label_ids(internal_polygons)
+
+
+@pytest.mark.parametrize('output_mesh_type', ['triangles', 'quads'])
+@pytest.mark.parametrize('scalars', ['labels', None])
+@pytest.mark.needs_vtk_version(9, 3, 0)
+def test_contour_labels_scalars_output_mesh_type(
+    labeled_image,
+    output_mesh_type,
+    scalars,
+):
+    # Determine expected output
+    if output_mesh_type == 'triangles' or output_mesh_type is None:
+        expected_celltype = pv.CellType.TRIANGLE
+    else:
+        assert output_mesh_type == 'quads'
+        expected_celltype = pv.CellType.QUAD
+
+    # Do test
+    mesh = labeled_image.contour_labels(
+        scalars=scalars,
+        output_mesh_type=output_mesh_type,
+    )
+    assert BOUNDARY_LABELS in mesh.cell_data
+    assert mesh.active_scalars_name == BOUNDARY_LABELS
+    assert all(cell.type == expected_celltype for cell in mesh.cell)
+    label_ids = _get_ids_with_background_boundary(mesh)
+    assert label_ids == [2, 5]
+
+
+def _remove_duplicate_points(polydata):
+    return polydata.clean(
+        point_merging=False,
+        lines_to_points=False,
+        polys_to_lines=False,
+        strips_to_polys=False,
+        inplace=False,
+    )
+
+
+@pytest.mark.parametrize(
+    'select_inputs',
+    [None, 2, 5, [2, 5]],
+    ids=['in_None', 'in_2', 'in_5', 'in_2_5'],
+)
+@pytest.mark.parametrize(
+    'select_outputs',
+    [None, 2, 5, [2, 5]],
+    ids=['out_None', 'out_2', 'out_5', 'out_2_5'],
+)
+@pytest.mark.parametrize('boundary_style', ['all', 'external'])  # , 'internal'],
```

Comment:
Please add a comment as to why we are skipping `internal` .

Reply:
This omission is probably an oversight, and the tests probably need an additional branch to accommodate this case (e.g. set the expected output). 

But, this case _is_ covered elsewhere with plotting tests (not included in this PR)...

Reply:
Should be fixed with [f5b2adb](https://github.com/pyvista/pyvista/pull/7000/commits/f5b2adb8b087e25fc5cb5cca11a4f4e2d6c13380)